### PR TITLE
Improve the handling of I/O port CF9

### PR DESCRIPTION
**Pull request by ZideChen0**
This patch series changes the implementation, so that in SOS, I/O requests to CF9, while in UOS, it's forwarded to device model and in partition mode, it's handled by the default handler outside of vpci.

Tracked-On: #1815
Signed-off-by: Zide Chen <zide.chen@intel.com>

### CODE REVIEW

**Comment by HuiHuangShi** [comment]
1816<br>No new violations to the coding guideline detected.

**Comment by HuiHuangShi** [comment]
1816<br>No new violations to the coding guideline detected.

**Comment by lijinxia** [comment]
@ZideChen0 Please rebase your folk branch and fix the conflicts.

File: hypervisor/dm/vpci/vpci.c [hunk 1]
```diff
@@ -30,97 +30,88 @@
...
+			vpci->ops->cfgwrite(vpci, pi->cached_bdf, pi->cached_reg + offset, bytes, val);
 		}
-	} else {
-		pr_err("Not PCI cfg data/addr port access!");
+		pci_cfg_clear_cache(pi);
 	}
 }
 
+static inline void register_pci_pio_handler(struct acrn_vm *vm, uint32_t pio_idx, uint32_t base, uint32_t nr_ports,
+	io_read_fn_t io_read_fn_ptr, io_write_fn_t io_write_fn_ptr)
```
Comment (dongyaozu) [inline hypervisor/dm/vpci/vpci.c]: can we remove this API ?

File: hypervisor/dm/vpci/vpci.c [hunk 2]
```diff
@@ -131,7 +122,12 @@ void vpci_init(struct acrn_vm *vm)
 #endif
 
 	if ((vpci->ops->init != NULL) && (vpci->ops->init(vm) == 0)) {
-		register_io_emulation_handler(vm, PCI_PIO_IDX, &pci_cfg_range, pci_cfg_io_read, pci_cfg_io_write);
+		register_pci_pio_handler(vm, PCI_CFGADDR_PIO_IDX, PCI_CONFIG_ADDR, 4U,
```
Comment (dongyaozu) [inline hypervisor/dm/vpci/vpci.c]: Can we use the original style here?

File: hypervisor/dm/vpci/vpci.c (hunk 1 above)
Comment (ZideChen0) [reply hypervisor/dm/vpci/vpci.c]: OK, I will remove it.

File: hypervisor/dm/vpci/vpci.c (hunk 2 above)
Comment (ZideChen0) [reply hypervisor/dm/vpci/vpci.c]: OK

**Comment by HuiHuangShi** [comment]
1816<br>No new violations to the coding guideline detected.

File: hypervisor/dm/vpci/vpci.c [hunk 3]
```diff
@@ -132,7 +112,18 @@ void vpci_init(struct acrn_vm *vm)
 #endif
 
 	if ((vpci->ops->init != NULL) && (vpci->ops->init(vm) == 0)) {
-		register_io_emulation_handler(vm, PCI_PIO_IDX, &pci_cfg_range, pci_cfg_io_read, pci_cfg_io_write);
+		pci_cfg_range.flags = IO_ATTR_RW;
+		pci_cfg_range.base = (uint16_t)PCI_CONFIG_ADDR;
+		pci_cfg_range.len = (uint16_t)4U;
```
Comment (dongyaozu) [inline hypervisor/dm/vpci/vpci.c]: please use the original style. use static array rather than code

File: hypervisor/dm/vpci/vpci.c [hunk 4]
```diff
@@ -132,7 +112,18 @@ void vpci_init(struct acrn_vm *vm)
...
+		register_io_emulation_handler(vm, PCI_CFGADDR_PIO_IDX, &pci_cfg_range,
+			pci_cfgaddr_io_read, pci_cfgaddr_io_write);
+
+		pci_cfg_range.flags = IO_ATTR_RW;
+		pci_cfg_range.base = (uint16_t)PCI_CONFIG_DATA;
+		pci_cfg_range.len = (uint16_t)4U;
```
Comment (dongyaozu) [inline hypervisor/dm/vpci/vpci.c]: ditto

File: hypervisor/dm/vpci/vpci.c (hunk 3 above)
Comment (ZideChen0) [reply hypervisor/dm/vpci/vpci.c]: OK

**Comment by HuiHuangShi** [comment]
1816<br>No new violations to the coding guideline detected.

**Pull request merged**